### PR TITLE
Update to curve25519-dalek-4.0.0-rc.1 

### PR DESCRIPTION
**Pull request by samdealy**
Updates to the latest published version of `curve25519-dalek`. 

NOTE: The code is not formatted with `cargo fmt` because the upstream repo doesn't use this formatting.

### CODE REVIEW

**Comment by samdealy** [comment]
Current dependencies on/for this PR:
* mobilecoin
  * **PR #4** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/x25519-dalek/4" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/x25519-dalek/4?utm_source=stack-comment).

File: .gitignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ Cargo.lock
 *.bak
 
 *.s
+.idea/
```
Comment (jcape) [inline .gitignore]:
```suggestion
```

File: tests/x25519_tests.rs [hunk 2]
```diff
@@ -21,7 +20,7 @@ fn byte_basepoint_matches_edwards_scalar_mul() {
 
         let result = x25519(scalar_bytes, X25519_BASEPOINT_BYTES);
 
-        let expected = (&ED25519_BASEPOINT_TABLE * &clamp_scalar(scalar_bytes))
+        let expected = (EdwardsPoint::mul_base( &clamp_scalar(scalar_bytes)))
```
Comment (jcape) [inline tests/x25519_tests.rs]:
```suggestion
        let expected = (EdwardsPoint::mul_base(&clamp_scalar(scalar_bytes)))
```

**Review by jcape — DISMISSED** [review]
LGTM, but please remove the `.idea` from `.gitignore`.

**Review by nick-mobilecoin — DISMISSED** [review]

**Review by nick-mobilecoin — APPROVED** [review]

**Pull request merged**